### DISCUSSION TITLE
Windows install: unresolved imports `libc::gethostname`, `libc::sockaddr_in`, `libc::_SC_HOST_NAME_MAX`

### DISCUSSION THREAD

**Discussion by l-7-l** [nxthat/nanocl#1081] [Q&A] [answered]
C:\Users\l-7-l\.cargo\registry\src\index.crates.io-6f17d22bba15001f\nanocl_utils-0.6.0\src\unix\network.rs:5:20
  |
5 | use libc::{c_char, gethostname, sockaddr_in, _SC_HOST_NAME_MAX};
  |                    ^^^^^^^^^^^  ^^^^^^^^^^^  ^^^^^^^^^^^^^^^^^ no `_SC_HOST_NAME_MAX` in the root
  |                    |            |
  |                    |            no `sockaddr_in` in the root
  |                    no `gethostname` in the root
  |
help: a similar name exists in the module
  |
5 | use libc::{c_char, getsockname, sockaddr_in, _SC_HOST_NAME_MAX};
  |                    ~~~~~~~~~~~
help: a similar name exists in the module
  |
5 | use libc::{c_char, gethostname, sockaddr, _SC_HOST_NAME_MAX};
  |                                 ~~~~~~~~

error[E0425]: cannot find function `getifaddrs` in crate `libc`
  --> C:\Users\l-7-l\.cargo\registry\src\index.crates.io-6f17d22bba15001f\nanocl_utils-0.6.0\src\unix\network.rs:40:28
   |
40 |   let ret = unsafe { libc::getifaddrs(ifaddrs.as_mut_ptr()) };
   |                            ^^^^^^^^^^ not found in `libc`

error[E0425]: cannot find function `freeifaddrs` in crate `libc`
  --> C:\Users\l-7-l\.cargo\registry\src\index.crates.io-6f17d22bba15001f\nanocl_utils-0.6.0\src\unix\network.rs:62:18
   |
62 |   unsafe { libc::freeifaddrs(ifaddrs) };
   |                  ^^^^^^^^^^^ not found in `libc`

**Comment by 0xle0ne** [comment, marked as answer]
Thx i'll try to fix this in next release

**Comment by 0xle0ne** [comment]
The new 0.16 version should be able to compile on windows!

**Reply by danielwernerio** [reply]
Very important to know for the community!
